### PR TITLE
DATA: add chairs and documentation.

### PR DESCRIPTION
**Pull request by YannDubs**
- added documentation for dataset
- added chairs

### CODE REVIEW

File: utils/datasets.py [hunk 1]
```diff
@@ -25,7 +27,19 @@ def get_img_size(dataset):
 
 def get_dataloaders(dataset, root=None, shuffle=True, pin_memory=True,
                     batch_size=128, **kwargs):
-    """A generic data loader"""
+    """A generic data loader
+
+    Parameters
+    ----------
+    dataset : {"mnist", "fashionMnist", "dsprites", "celeba", "chairs"}
```
Comment (linesd) [inline utils/datasets.py]: Should this be "fashion" instead of "fashionMnist" ?

File: utils/datasets.py [hunk 2]
```diff
@@ -67,14 +94,20 @@ def __len__(self):
...
         Return
         ------
         sample : torch.Tensor
             Tensor in [0.,1.] of shape `img_size`.
         """
         pass
 
+    @abc.abstractmethod
+    def downlaod(self):
+        """Dowanload the dataset. """
```
Comment (linesd) [inline utils/datasets.py]: "download" spelled incorrectly in both the function name and the docstring

File: utils/datasets.py [hunk 3]
```diff
@@ -67,14 +94,20 @@ def __len__(self):
...
+        pass
+
```
Comment (linesd) [inline utils/datasets.py]: Is this function just a placeholder at the moment?

Comment (YannDubs) [reply utils/datasets.py]: `@abc.abstractmethod`means that all objects that inherit from this class have to define that method (to be more precise: it's not possible to instantiate an object with an abstract method)

So because no other classes have a method called (with mispelling) `downlaod`, it should break the code. But I forgot that when you define an abstract method you shouldn't use `pass`. Because `pass` is a special keyword that defined a function without defining it. So actually the code doesn't break because they all have a method called `downlaod` which does nothing. 

TL;DR: thanks I need to remove `pass` and change the typo

Comment (linesd) [reply utils/datasets.py]: Let me know when you've done changes and I'll merge

Comment (YannDubs) [reply utils/datasets.py]: actually, the `pass` was correct (sorry I commented before testing haha) it was `abc.ABC` missing. SOlved and merge
